src/core/packages/saved-objects/api-server-internal/src/lib/point_in_time_finder.ts

```diff
@@ -180,4 +188,26 @@ export class PointInTimeFinder<T = unknown, A = unknown>
     }
     return res.saved_objects[res.saved_objects.length - 1].sort;
   }
+
+  /**
+   * Safely extracts error message and status information from various error types
+   * @param error - The error object that could be any type
+   * @returns A safe string representation of the error
+   */
+  private getErrorMessage(error: unknown): string {
+    if (Boom.isBoom(error)) {
+      return error.output.payload.statusCode
+        ? `${error.output.payload.statusCode} ${error.message}`
+        : error.message;
```

Comment:
Are we 100% sure we won't be leaking any PII or sensitive info through the message? Rather us [decorated ES errors] (https://github.com/elastic/kibana/blob/932684b3d21549dd664d039c6ffa26b9d508c7ff/src/core/packages/saved-objects/api-server-internal/src/lib/utils/decorate_es_error.ts#L24)